lib/app/group.js

```diff
@@ -20,9 +20,23 @@ var membership = [
 ];
 
 module.exports = function(mediator) {
-  var groupStore = new ArrayStore('group', groups);
-  groupStore.listen('cloud:', mediator);
+  var groupStore = new SimpleStore('group');
+  groupStore.list().then(function(res) {
 
-  var membershipStore = new ArrayStore('membership', membership);
-  membershipStore.listen('cloud:', mediator);
+    if (!res || res.length === 0) {
+      groupStore.init(groups);
+    }
+
+    groupStore.listen(':cloud', mediator);
+  });
+
+  var membershipStore = new SimpleStore('membership');
```

Comment:
there is duplication of functionality for each module. Create module store-init and then apply() on different modules